services/cd-service/pkg/repository/transformer.go

```diff
@@ -365,8 +366,25 @@ type transformerRunner struct {
 	DeletedRootApps []RootApp
 }
 
+var isolatedTransformersLock sync.RWMutex
+var isolatedTransformerNames = []string{"UndeployApplication", "DeleteEnvFromApp", "DeleteEnvironment"}
+
 func (r *transformerRunner) Execute(ctx context.Context, t Transformer, transaction *sql.Tx) error {
 	r.Stack = append(r.Stack, nil)
+	tranformerTypeName := reflect.TypeOf(t).Name()
+	isIsolated := false
+	for _, isolatedTransformerName := range isolatedTransformerNames {
+		if isolatedTransformerName == tranformerTypeName {
+			isIsolated = true
+		}
+	}
```

Comment:
I think this would work for now, but it's not future proof.
If we scale up the cd-service to 2 pods, then this solution is not effective anymore.
At minimum, we need a story to come up with a long term solution.